### PR TITLE
* SCP-2860 Add support for zipped unsorted MTX files

### PR DESCRIPTION
**Pull request by knapii-developments**
This PR adds support for unsorted compressed MTX files.

### CODE REVIEW

**Review by bistline — APPROVED** [review]
I'm assuming the reason you're actually unzipping the matrix is so that it can be sorted?  If so this looks good.

**Review by eweitz — CHANGES_REQUESTED** [review]
I'm glad this new support is getting refined!

I found a blocker:  see comment beginning "[This has a subtle bug](https://github.com/broadinstitute/scp-ingest-pipeline/pull/162#discussion_r516096159)".  I also suggested some readability improvements.

File: ingest/expression_files/mtx.py [hunk 1]
```diff
@@ -214,20 +239,13 @@ def sort_mtx(file_path, mtx_file_handler: IO) -> str:
          ----------
             new_file_path (str) : Full path of newly sorted MTX file
         """
-        file_name = ntpath.split(file_path)[1]
+        file_name = ntpath.split(file_path)[1][:-3]
```
Comment (eweitz) [inline ingest/expression_files/mtx.py:242]: This has a subtle bug: `file_name` retains a period.  The [robust approach you use elsewhere](https://github.com/broadinstitute/scp-ingest-pipeline/pull/162/files#diff-523e80ea0f210d08dda74bd7279f507ef450e6aa75d444aac8bee2f6bbed2895R95) seems preferable.

Bug demo:
```
wm10d-4e7:single_cell_portal_core eweitz$ python3
Python 3.7.7 (default, Mar 10 2020, 15:43:03) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ntpath
>>> file_path = '/path/to/file.txt'
>>> ntpath.split(file_path)[1][:-3]
'file.'
```

Applying your other, robust approach:

```suggestion
        file_name = os.path.basename(file_path)
        file_name = os.path.splitext(base)[0]
```

File: ingest/ingest_files.py [hunk 2]
```diff
@@ -135,7 +135,7 @@ def resolve_path(self, file_path):
             file_path = self.download_from_bucket(file_path)
             self.local_file_path = file_path
         # Remove BOM with encoding ='utf - 8 - sig'
-        if self.is_gzip_file:
+        if self.get_file_type(file_path)[1] == "gzip":
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply ingest/ingest_files.py:138]: 9b6af33

File: ingest/expression_files/mtx.py [hunk 3]
```diff
@@ -84,32 +84,57 @@ def check_valid(
...
-        """Checks if a file is sorted by gene index"""
+    def get_gene_expression_data(file_path: str, file_handler):
+        """Pipes expression data out for zipped and unzipped files"""
+        std_in = file_path
         file_size = os.path.getsize(file_path)
         if file_size == 0:
             raise ValueError(f"{file_path} is empty: " + str(file_size))
-        start_idx: int = MTXIngestor.get_data_start_line_number(file_handler)
+
+        if IngestFiles.get_file_type(file_path)[1] == "gzip":
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply ingest/expression_files/mtx.py:94]: 9b6af33

File: ingest/ingest_files.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (knapii-developments) [reply ingest/ingest_files.py]: Tests for this 25e8ae7

**Review by eweitz — APPROVED** [review]

**Pull request merged**